### PR TITLE
Add a task for plotting the sea surface height vs. obs

### PR DESCRIPTION
**Pull request by xylar**
A task for plotting the SSH vs. observations from AVISO.

### CODE REVIEW

**Comment by xylar** [comment]
This is a revival of #279, since it seems like a task to compare SSH with obs. might, in fact, be useful.  Mark wants to include one such plot in the MPAS-O paper.

**Comment by xylar** [comment]
@vanroekel and @milenaveneziani, could you each do a quick run with `--generate=climatologyMapSSH` to make sure you can, indeed, generate an SSH plot?  I could use some advice on the range of the color bars (and would be open to changing the color map if need be).

I have made sure the necessary observations are on Edison but I'm not sure if they have been copied over to other machines.  Please let me know if you find them to be missing.

**Comment by xylar** [comment]
@mark-petersen, I know you're busy with the paper so all I would want from you is verification that the figure looks like what you were after, once we have an example plot posted here.

**Comment by xylar** [comment]
## Testing

I ran with both the QU240 test case on my laptop and the following EC60to30v3wLI test case:
http://portal.nersc.gov/project/m2833/mpas_analysis_output/20180209.GMPAS-IAF.T62_oEC60to30v3wLI.cori-knl.afterSalinityFix/ocean/index.html#ssh

Please comment on the color map and the range.  I will be fixing the values near zero similar to #311.

**Comment by vanroekel** [comment]
@xylar tested on 60to30 and 30to10 on anvil and things worked great.  Examples below
sshaviso_20180208 gmpas-iaf t62_orrs18v3 anvil_ann_years0001-0007
sshaviso_20180123 test-low-restoring gmpas-iaf t62_oecv3 anvil_ann_years0001-0020

I do think a tighter range would be good.  It is hard to see detail in this range.  Perhaps also a diverging colormap like the red blue one might make features more visible?

**Comment by milenaveneziani** [comment]
I have a suggestion for a colormap. I quite like the Spectral_r, with the following color indeces:
[0, 20, 40, 80, 100, 120, 160, 180, 200, 240, 255].
And we could try the following levels:
[-2.4, -2, -1.6, -1.2, -0.8, -0.4, 0, 0.4, 0.8, 1.2]

File: mpas_analysis/ocean/climatology_map_ssh.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                # make a new subtask for this season and comparison grid
+                subtask = PlotClimatologyMapSubtask(self, season,
+                                                    comparisonGridName,
+                                                    remapClimatologySubtask,
+                                                    remapObservationsSubtask,
+                                                    refConfig)
+
+                subtask.set_plot_info(
+                        outFileLabel=outFileLabel,
+                        fieldNameInTitle='SST',
```
Comment (vanroekel) [inline mpas_analysis/ocean/climatology_map_ssh.py]: SSH here right?

**Comment by milenaveneziani** [comment]
ah, also, I noticed we need to adjust title and units to 'SSH' and 'cm' (I suppose).

File: mpas_analysis/ocean/climatology_map_ssh.py (hunk 1 above)
Comment (xylar) [reply mpas_analysis/ocean/climatology_map_ssh.py]: Oops, yes.

**Comment by vanroekel** [comment]
here is a version with the RdBu color map.  It is a bit clearer to my eye
sshaviso_20180123 test-low-restoring gmpas-iaf t62_oecv3 anvil_ann_years0001-0020

**Comment by vanroekel** [comment]
oops, need to fix the colors.  I see negative values with red.

**Comment by xylar** [comment]
@milenaveneziani, good catch.  Units are meters.

As far as color maps, I'll try your suggestion.  Spectral_r seems like a good option and I don't think we've used it for other plots.

**Comment by milenaveneziani** [comment]
@vanroekel's choice is nice too (to make it symmetric wrt 0 we need to change the levels, which may not be great in this case because SSH is way more negative than positive).
Here is an example of using spectral_r for salinity:
saltdepth0000_ginsea_b1850c5_acmev0_highres_0121-10

**Comment by vanroekel** [comment]
I think spectral_r is nice for this too.  Here is an example
sshaviso_20180123 test-low-restoring gmpas-iaf t62_oecv3 anvil_jas_years0001-0020

**Comment by xylar** [comment]
Great, I think we're on to something.  My plots look very similar to Luke's above.  Check the link under testing again.

**Comment by milenaveneziani** [comment]
I was looking at the Centurioni paper about their Mean Dynamic Height product (which at some point we discussed we would use for this, but we never downloaded it), and I thought this was a nice picture:

screen shot 2018-02-21 at 10 15 39 pm

perhaps at some point we could add the 10cm contours to ours?

**Comment by milenaveneziani** [comment]
yes, the new ones look nice.

**Comment by xylar** [comment]
Wow, @milenaveneziani, I agree that's quite a nice looking figure.  That's probably a custom color map that we could easily duplicate (I added a couple of custom color maps as well in the plotting module).  Given that adding contours would be a non-trivial modification to plotting, I think we should leave that for a follow-up PR.  Perhaps after the paper, given the very tight timeline.

**Comment by xylar** [comment]
@mark-petersen, I'm fine with switching to cm if that's standard.  The standard is dumb in this case but it's not my place to try to force a new standard... :-p

**Comment by xylar** [comment]
Here is a plot with units of cm instead of m:
image

**Pull request merged**